src/ray/core_worker/transport/direct_actor_transport.h

```diff
@@ -478,6 +486,14 @@ class CoreWorkerDirectTaskReceiver {
   std::unordered_map<TaskID,
                      std::pair<SchedulingQueueTag, std::unique_ptr<SchedulingQueue>>>
       scheduling_queue_;
+
+  /// Protects access to the counters below.
+  absl::Mutex mu_;
+
+  /// Queue to register request accept callback.
+  /// This is used to complete request accept callback.
+  /// NOTE: This must be guarded when used by core worker.
```

Comment:
Use thread annotations for this. See `core_worker.h`

Reply:
We will remove this map if we combine to callback saving layers into one!